app/javascript/gobierto_data/webapp/components/sets/data/SQLEditorResults.vue

```diff
@@ -59,6 +115,40 @@ export default {
       const config = this.$refs.viewer.getConfig()
 
       this.$root.$emit("storeCurrentVisualization", config, opts);
+      this.removeLabelBtn = true
+      this.isVisualizationModified = false
+    },
+    resetViz() {
+      const hidePerspective = "none"
+
+      this.showVisualize = true
+      this.perspectiveChanged = false
+      this.removeLabelBtn = false
+      this.showResetViz = false
+      this.typeChart = 'hypergrid'
+      this.isVisualizationModified = false
+
+      this.$refs.viewer.enableDisabledPerspective(hidePerspective);
+      this.$refs.viewer.setColumns();
+    },
+    showChart() {
+      const showPerspective = "flex"
+
+      this.showVisualization = true
+      this.$refs.viewer.enableDisabledPerspective(showPerspective);
+    },
```

Comment:
What are this Visualizations functions doing enclosed here? You're messing with the logic. SQLEditorResults does know nothing about what Visualizations does. The parent just "inform" the children about showing/hiding, nothing else.
These functions does not belong here.